### PR TITLE
refactor(eudr): extract overage tier constants and helpers (#793)

### LINKED ISSUE #793: [refactor] Replace cast() with runtime validation in orchestrator + move EUDR tier thresholds to constants
## Problem

Two related code-hygiene issues identified in the C4 Architecture Deep Review:

### 1. Excessive `cast()` use in orchestrator phases

`blueprints/pipeline/orchestrator.py` uses `typing.cast(...)` extensively to silence type errors when accessing fields from dynamically-shaped dicts returned by activity functions:

```python
ingest_summary = cast(dict[str, Any], context.task_all([...]).get(0))
acquire_summary = cast(dict[str, Any], ingest_summary.get("acquisition"))
fulfilment_summary = cast(dict[str, Any], acquire_summary.get("fulfilment"))
```

`cast()` is a runtime no-op — it tells the type checker to trust us, but provides zero protection if the activity returns an unexpected shape. If an upstream activity is changed and forgets to populate `acquisition`, we get an `AttributeError` deep inside a phase function rather than a clear validation error at the boundary.

### 2. Hardcoded EUDR tier thresholds

`blueprints/eudr.py` `_eudr_usage_payload` (around lines 200–230) hardcodes tier prices and quotas inline:

```python
if quota_used < 100:
    unit_price_eur = 2.50
elif quota_used < 500:
    unit_price_eur = 1.80
else:
    unit_price_eur = 1.20
```

Magic numbers (100, 500, 2.50, 1.80, 1.20) make it:
- Hard to update pricing (need to grep across multiple files)
- Hard to audit (no single source of truth for "what does our pricing look like?")
- Hard to test edge cases (no parametric tier definition)

## Why it matters

- **Type safety theatre**: `cast()` gives false confidence in payload shapes
- **Pricing drift risk**: tier numbers in `eudr.py` could diverge from billing module / Stripe / docs
- **Audit burden**: Every pricing review requires reading code instead of a table

### Part 1: Replace `cast()` with runtime validation

Add small Pydantic models (or `TypedDict` + a `validate` helper) for activity outputs:

```python
# treesight/pipeline/contracts.py
class IngestionResult(BaseModel):
    submission_id: str
    parsed_features: list[dict[str, Any]]
    acquisition: AcquisitionPayload  # nested model

# orchestrator.py
ingest_summary = IngestionResult.model_validate(
    context.task_all([...]).get(0)
)
acquire_summary = ingest_summary.acquisition  # type-safe attribute access
```

Or, lighter weight, a `validate_phase_output(name: str, data: Any, required_keys: tuple[str, ...]) -> dict` helper that fails fast with a clear error.

### Part 2: Move EUDR tiers to constants

Add to `treesight/constants.py`:

```python
EUDR_TIERS: tuple[EudrTier, ...] = (
    EudrTier(min_units=0,   unit_price_eur=Decimal("2.50"), label="starter"),
    EudrTier(min_units=100, unit_price_eur=Decimal("1.80"), label="growth"),
    EudrTier(min_units=500, unit_price_eur=Decimal("1.20"), label="scale"),
)

def eudr_unit_price(quota_used: int) -> Decimal:
    """Return the marginal price for the given usage level."""
    return next(
        tier.unit_price_eur
        for tier in reversed(EUDR_TIERS)
        if quota_used >= tier.min_units
    )
```

`_eudr_usage_payload` then calls `eudr_unit_price(quota_used)` — single source of truth.

## Acceptance criteria

- At least 3 `cast()` calls in `orchestrator.py` replaced with runtime validation
- `EUDR_TIERS` constant + `eudr_unit_price()` helper added to `treesight/constants.py` (or `treesight/billing/eudr_tiers.py`)
- `_eudr_usage_payload` and any other tier-aware code updated to consume the helper
- Unit tests cover: each tier boundary (0, 99, 100, 499, 500, 1000), runtime validation rejects malformed payloads
- No behaviour change in pricing for existing usage levels (regression tests pass)

## Discovered in

C4 Architecture Deep Review (PR #NNN). Sections:
- "Code Review → Idiomaticity → cast() overuse"
- "Code Review → Magic Numbers → EUDR tier thresholds"

## Effort

Small — both changes are localised. Can be done in one PR or split if reviewers prefer.

## Labels

`refactor`, `code-quality`, `eudr`, `pipeline`, `P3`

### PR DESCRIPTION
**Pull request by Hardcoreprawn**
## Summary

Resolves the EUDR magic-numbers portion of #793 by extracting graduated overage tier pricing into `treesight/constants.py` and adding `eudr_unit_price_gbp()` and `eudr_next_tier()` helpers in `treesight/security/eudr_billing.py`.

## Changes

**`treesight/constants.py`** — new constants under existing EUDR Billing section:
- `EUDR_OVERAGE_BASE_RATE_GBP = 3.0` — replaces the bare `3.0` multiplier
- `EUDR_OVERAGE_TIERS = ((100, 2.50), (500, 1.80))` — replaces the inline tuple

**`treesight/security/eudr_billing.py`** — two new helpers:
- `eudr_unit_price_gbp(period_parcels_used: int) -> float` — marginal £/parcel rate at a given usage level
- `eudr_next_tier(period_parcels_used: int) -> tuple[int, float] | tuple[None, None]` — next-tier guidance for upsell prompts

**`blueprints/eudr.py`** — `_eudr_usage_payload` now consumes both helpers. No behaviour change.

**`tests/test_eudr_billing.py`** — 13 new parameterised tests covering all tier boundaries.

## Behaviour preservation

All existing tests pass (`tests/test_eudr_billing_endpoints.py`, `tests/test_eudr_billing.py`). New tests confirm pricing at every boundary (0, 50, 99, 100, 250, 499, 500, 5000) matches the previous inline values exactly.

## Scope split — `cast()` work deferred

Issue #793 originally bundled "replace `cast()` with runtime validation" with the EUDR work. After investigating, the 16 `cast()` calls in `blueprints/pipeline/orchestrator.py` are bridging **Durable Functions yield typing limitations**, not lying about dict shapes:

```python
features = cast("Any", (yield context.call_activity("parse_kml", inp)))
```

The `yield` returns `Any` because mypy can't infer Durable activity outputs through the generator protocol. Replacing with naive runtime validation (e.g. Pydantic on every yield) would be invasive and risk regressions on partial/missing data that the orchestrator currently degrades through.

That work needs its own design discussion (typed activity contracts, possibly `TypedDict` per-activity, or a thin `validate_activity_output()` boundary check). Splitting into a separate issue keeps this PR small and reviewable.

→ Filing follow-up issue for the `cast()` portion; #793 will close on merge of this PR.

## Validation

- `uv run ruff check` — clean
- `uv run ruff format --check` — clean
- Targeted tests: 73 passed (`test_eudr_billing.py`, `test_eudr_billing_endpoints.py`, `test_constants.py`)
- Full suite: 1620 passed; 7 pre-existing failures in `test_demo_tier.py`, `test_helpers.py`, `test_ops.py` are all the same `pytest-asyncio` issue tracked in #786 — confirmed pre-existing on `main` via stash + re-run.

## Risk

Low. Pure refactor — no behaviour change at any tested usage level. Pricing calculations exercised by both new boundary tests and existing endpoint tests.

Closes #793 (after follow-up issue is filed for `cast()` split).

### CODE REVIEW

File: blueprints/eudr.py [hunk 1]
```diff
@@ -241,8 +241,8 @@ def _fetch_org_run_records(user_id: str, limit: int = 250) -> list[dict]:
 
 
 def _eudr_usage_payload(user_id: str) -> dict:
-    from treesight.constants import EUDR_INCLUDED_PARCELS
-    from treesight.security.eudr_billing import get_eudr_billing_status
+    from treesight.constants import EUDR_INCLUDED_PARCELS, EUDR_OVERAGE_BASE_RATE_GBP
+    from treesight.security.eudr_billing import eudr_next_tier, get_eudr_billing_status
     from treesight.security.orgs import get_user_org
```
[the remark this answers is not in the document]
Comment (Hardcoreprawn) [reply blueprints/eudr.py]: @copilot apply changes based on [this feedback](https://github.com/Hardcoreprawn/azure-workflow-for-kml-satellite/pull/794#discussion_r3221973599)
